analyses/tmb-compare-tcga/TMB_d3b_code/code/02_cumulative_freq_TMBplot.py

```diff
@@ -0,0 +1,94 @@
+##  This script takes in TMB scores outfile and calculates cumulative distribution
+#   function plot.
+# INputs  include the tmb  scores text file, outfile name and min number of samples to be plotted
+#                   (no need to use any extensions for out file, just the name)
+#   Somtimes MAF can have very few samples under each histology and those histologies can be filtered
+#       out of the plot using the min samples feature
+
+# The TMB  text  should have the following column headers -
+#   Samplename experimental_strategy disease count bedlength  TMB
+# Use results from 01_calculate_tmb_targetflexible.py script
+
+####### Author: Teja Koganti ##################
+# python3 02_cumulative_freq_TMBplot.py  \
+#   -i ../output/pbta-snv-consensus.tmb.txt \
+#   -o ../output/pbta-snv-consensus \
+#   -s 10
+
+import argparse
+import subprocess
+import sys
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-t", "--tmb_scores", required=True, help="file with TMB scores")
+parser.add_argument(
+    "-o", "--outfilename", required=True, help="Name of the out plot, no extension"
+)
+parser.add_argument(
+    "-s",
+    "--minsamplestoplot",
+    required=True,
+    help="Minimum samples from each histology/disease to plot",
+)
+args = parser.parse_args()
+
+######### Preparing inputs files and disease list ###########
+# Loading TMB scres text file into a  dataframe
+tmbfile = pd.read_csv(args.tmb_scores, sep="\t")
+
+# Choosing disease list to plot
+#tmbfile["new_TMB"] = tmbfile["TMB"]+1
+value_to_add = np.min(tmbfile[tmbfile["TMB"] != 0]["TMB"])*0.8
+tmbfile["new_TMB"] = tmbfile["TMB"].replace({0: value_to_add})
+tmbfile = tmbfile.sort_values(by=['new_TMB'])
+
+
+disease_number = tmbfile.groupby("cohort").count()["new_TMB"]
```

Comment:
Do we want to plot by cohort or disease?

Reply:
We were using both for PNOC003, but I will change this to disease for this code